api/src/main/java/org/apache/gravitino/stats/SupportsPartitionStatistics.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.stats;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.gravitino.exceptions.UnmodifiableStatisticException;
+
+/** SupportsPartitionStatistics provides methods to list and update statistics for partitions. */
+interface SupportsPartitionStatistics {
+
+  /**
+   * Lists statistics for partitions from one partition name to another partition name.
+   *
+   * @param fromPartitionName Optional partition name to start listing from, inclusive.
+   * @param toPartitionName Optional partition name to end listing at, exclusive.
+   * @throws IllegalArgumentException if fromPartitionName and toPartitionName are empty at the same
+   *     time or the toPartitionName partition key must be greater than fromPartitionName partition
+   *     key.
+   * @return a map where the key is the partition name and the value is a list of statistics
+   */
+  Map<String, List<Statistic>> listStatistics(
+      Optional<String> fromPartitionName, Optional<String> toPartitionName)
+      throws IllegalArgumentException;
```

Comment:
It is better if we could impose some restrictions on the input parameters to avoid user to choose `Optional` for both from and to. Can you refer to other projects to see how they implement this?